crates/iota-data-ingestion/src/workers/kv_store.rs

```diff
@@ -152,32 +144,53 @@ impl KVStoreWorker {
         Ok(())
     }
 
-    async fn upload_blob<V: Serialize + std::marker::Send>(
+    /// Uploads checkpoint contents to storage, with automatic fallback from
+    /// DynamoDB to S3.
+    ///
+    /// This function attempts to store checkpoint contents in DynamoDB first.
+    /// If that fails (typically due to size constraints), it automatically
+    /// falls back to uploading the contents to S3 instead.
+    async fn upload_checkpoint_contents<V: Serialize + std::marker::Send>(
         &self,
         key: Vec<u8>,
         value: V,
     ) -> anyhow::Result<()> {
-        let body = bcs::to_bytes(value.borrow())?.into();
-        self.s3_client
-            .put_object()
-            .bucket(self.bucket_name.clone())
-            .key(base64_url::encode(&key))
-            .body(body)
+        let bcs_bytes = bcs::to_bytes(value.borrow())?;
+
+        let attributes = HashMap::from([
+            (
+                "digest".to_string(),
+                AttributeValue::B(Blob::new(key.clone())),
+            ),
+            (
+                "type".to_string(),
+                AttributeValue::S(ItemType::CheckpointContents.to_string()),
+            ),
+            (
+                "bcs".to_string(),
+                AttributeValue::B(Blob::new(bcs_bytes.clone())),
+            ),
+        ]);
+
+        let res = self
+            .dynamo_client
+            .put_item()
+            .table_name(&self.table_name)
```

Comment:
Just for my understanding, I wonder why table name doesn't depend on object type. What do we use as table name?

Reply:
The table name is gathered from a config file, and it can be any string. For example this is the table name gathered form the config: https://github.com/iotaledger/iota/blob/develop/dev-tools/iota-data-ingestion/config/kv_store_config.yaml#L57

Currently we're using a single-table design pattern where all different object types are stored in the same DynamoDB table but with a `type` attribute that identifies each item's type (`tx` for transactions, `fx` for effects, etc.). This is a recommended pattern for DynamoDB that improves efficiency and simplifies operations, while still allowing us to query for specific object types.